marshal/consumer.go

```diff
@@ -132,9 +138,18 @@ func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*
 					}
 					c.claims[topic][partID] = newClaim(
 						topic, partID, c.marshal, c.messages, options)
+
+					// update topic claims
+					if options.ClaimEntireTopic && partID == 0 {
+						c.claimedTopics[topic] = true
+					}
 				}
 			}
 		}
+		// send topic claim notification
+		if options.ClaimEntireTopic {
+			c.topicClaimsChan <- c.claimedTopics
```

Comment:
This sends a pointer to the map which means that the receiver (since they can get it directly) could do bad things to our internal structure such as mutating it while we're looking at it later to check for differences.